### PR TITLE
In TA and Sectors, inferred relationships located-at should not have a button "unlink / delete" (#issue/4220)

### LINKED ISSUE #4220: In TA, inferred relationships located-at should not have a button "unlink / delete"
## Bug

In Threat Actor (all) => Overview:

Quick located-at is displaying the button "unlink" on inferred relationships (no effect).

image

### PR DESCRIPTION
**Pull request by SarahBocognano**
### Related issues
 
* https://github.com/OpenCTI-Platform/opencti/issues/4220

(Don't forget to enable rules propagation for located-at and part-of to repro cases)

### CODE REVIEW

**Comment by marieflorescontact** [comment]
locally tested. 👌

2 observations that are probably not related to this bug fix:
- fully refresh is required to display/delete inferred relationships when adding a relationship via the quick `add` button. 
- tooltip missing on the quick `add` button and on the `unlink / delete` button .


@SarahBocognano : Good call on the tooltip missing ! And you are right, the fully refresh is still required in the app to display/delete inferred relationships. That behavior will not be change on this issue.

@SarahBocognano : @marieflorescontact After investigation, there is no 'add' button or' unlike / delete' button who have a tooltip on the platform. So to not change the global behavior, I will not add them.

File: opencti-platform/opencti-front/src/private/components/entities/sectors/SectorSubSectors.jsx
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/entities/sectors/SectorSubSectors.jsx]: Where is it used `isSubSector` ?

File: opencti-platform/opencti-graphql/src/database/middleware.js
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/database/middleware.js]: could it happen that there are multiple relations for the same target ? since we are in a batch method, we return results for a list of entities. So we could have entityA that has relations, one inferred with targetA, and entityB that also has a relation with targetA that is not inferred, so what would happen ?

**Comment by Archidoit** [comment]
Don't know if it's a good idea, but maybe we can add an 'inferred' indication/icon instead of the removed 'unlink' button, to indicate why we can't manage the relation (concern the 'reg1' line in the screen short below):
image

@SarahBocognano : It could be a good idea to add an icon like in the "last created relationship" component where we display a particular icon for inferred relationship with a tooltip indicating why. But right now, this behavior is not present in other part of the platform (in IntrusionSet for instance) so I'm not sure if we should do it now. What is your thought on that @Jipegien ?

**Comment by Jipegien** [comment]
> Don't know if it's a good idea, but maybe we can add an 'inferred' indication/icon instead of the removed 'unlink' button, to indicate why we can't manage the relation (concern the 'reg1' line in the screen short below): image
> 
> @SarahBocognano : It could be a good idea to add an icon like in the "last created relationship" component where we display a particular icon for inferred relationship with a tooltip indicating why. But right now, this behavior is not present in other part of the platform (in IntrusionSet for instance) so I'm not sure if we should do it now. What is your thought on that @Jipegien ?

Same comportment for TA and IS. For IntrusionSet, it is not a "located-at" relationship but a "originates-from". For having the same comportment, we need to add an inference rule to propagates the originates-from to parents (region).

**Comment by RomuDeuxfois** [comment]
Hey,
What is the status of this work ?

**Comment by Kedae** [comment]
> Hey, What is the status of this work ?

Waiting for review

File: opencti-platform/opencti-front/src/private/components/entities/sectors/SectorSubSectors.jsx
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/entities/sectors/SectorSubSectors.jsx]: Instead of using a ternary with an empty React Fragment you can wrap `ListItemSecondaryAction` with a simple condition `!isInferred` like this:
```javascript
{!isInferred && (
  <ListItemSecondaryAction>
    <Security needs={[KNOWLEDGE_KNUPDATE]}>
      <IconButton
        aria-label="Remove"
        onClick={this.removeSubSector.bind(this, subSectorEdge)}
        size="large"
      >
        <LinkOff />
      </IconButton>
    </Security>
  </ListItemSecondaryAction>
)}
``` 

Same for the 2 other components

File: opencti-platform/opencti-graphql/src/modules/case/case-rfi/case-rfi.graphql
Comment (richard-julien) [inline opencti-platform/opencti-graphql/src/modules/case/case-rfi/case-rfi.graphql]: Not sure about the placement of this. We need to check if it could be more interesting to put the information in the edge type

File: opencti-platform/opencti-graphql/src/database/middleware.js
Comment (richard-julien) [inline opencti-platform/opencti-graphql/src/database/middleware.js]: I need more information about why we need to filter information here

**Review by richard-julien — CHANGES_REQUESTED** [review]

**Comment by SarahBocognano** [comment]
Update: Changes requested have been made. Waiting for another review

**Comment by SamuelHassine** [comment]
@Kedae can we push a little on this one after 4 months?

**Review by richard-julien — APPROVED** [review]

**Pull request merged**
